src/main/java/uk/ac/ic/wlgitbridge/bridge/lock/ProjectLockFactory.java

```diff
@@ -0,0 +1,6 @@
+package uk.ac.ic.wlgitbridge.bridge.lock;
+
+public interface ProjectLockFactory {
```

Comment:
<3 factory pattern. ;-)

But looking at it, I think that in this case we don't actually need a `ProjectLockFactory`. Because we have `ProjectContextFactory`, and these locks are created from within the subclasses contexts, we can just `new` them from the context themselves without using a factory.

I think this gets clearer if we move the `HashMap` to a static member in the `InProcessLock` class as we can just `new` those in the same way as the postgres ones.

That would let us remove this whole commit (and the three `LockFactory` classes) completely.